pkg/platform/packet/packet.go

```diff
@@ -525,6 +526,37 @@ func (c *config) validateOSVersion() hcl.Diagnostics {
 	return diagnostics
 }
 
+// checkWorkerPoolLabelsAndTaints verifies that all worker pool labels
+// and taints are in correct format. Neither key nor value of labels
+// and taints map should have empty space in them.
+func (c *config) checkWorkerPoolLabelsAndTaints() hcl.Diagnostics {
```

Comment:
I'd extend `platform.WorkerPool` interface and add generic validation for it so we don't duplicate code for each platform.